### DISCUSSION TITLE
[HELP] how to keep session when reload browser after call

### DISCUSSION THREAD

**Discussion by genffy** [versatica/JsSIP#884] [Q&A]
There is a scenario in our business: During a call, if the browser is refreshed or closed, and the page is reloaded (within a limited time), we hope that the current call can continue. May I ask how to achieve this?

I have read the source code and tried to operate through the instance on the session. However, all of these operations are after the newRTCSession, and it is not possible to initialize it based on some existing parameters.

**Comment by Subhan00** [comment]
Hi @genffy, I was facing the same issue — during an active call, if I refreshed the browser, the session would get terminated on my side, but the call would still remain active on the server. I managed to resolve this using the SIP Replaces header.

Here’s how I handled it:

Before Refresh: I stored the following session details in localStorage before the page unloads:

1. callId
2. fromTag
3. toTag
4. remoteURI

After Refresh: Once the page reloads and the UA is registered again, I retrieved those values from localStorage and used them to build the Replaces header like this:
const replacesHeader = `Replaces: ${callId};to-tag=${toTag};from-tag=${fromTag}`;

Then I initiated a new call to the same remote URI and passed this header in the extraHeaders of the call options.

This allowed me to resume the existing call session after the page was refreshed. Just make sure your server supports the Replaces header and the original call is still alive.

Hope this helps!

**Reply by svsubith** [reply]
Hi @Subhan00, May I know the server used in your scenario?
